source/tutorials/kubernetes.rst

```diff
@@ -180,71 +350,143 @@ Configure and run CRI-O + kata-runtime
+   mkdir -p $HOME/.kube
+   sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+   sudo chown $(id -u):$(id -g) $HOME/.kube/config
 
-.. code-block:: bash
+   ...
 
-   kubeadm join <master-ip>:<master-port> --token <token> --discovery-token-ca-cert-hash <hash>
+   You can now join any number of machines by running the following on each node
+   as root:
 
+   kubeadm join <control-plane-host>:<control-plane-port> --token <token> --discovery-token-ca-cert-hash sha256:<hash>
 
-**Congratulations!**
 
-You've successfully installed and set up Kubernetes in |CL| using CRI-O and kata-runtime. You are now ready to follow on-screen instructions to deploy a pod network to the cluster and join worker nodes with the displayed token and IP information.
+With the first node of the cluster setup, you can continue expanding the
+cluster with additional nodes and start deploying containerized applications.
+For further information on using Kubernetes, see `Related topics`_. 
 
-Related topics
-**************
+.. note:: 
 
-Read the Kubernetes documentation to learn more about:
+   By default, the master node does not run any pods for security reasons. To
+   setup a single-node cluster and  allow the master node to also run pods,
+   see the Kubernetes documentation on `control plane node isolation
+   <https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#control-plane-node-isolation>`_.
 
-*  Deploying Kubernetes with a `cloud-native-setup`_
 
-*  :ref:`Kubernetes best practices <kubernetes-bp>`
+Reference
+*********
 
-* `Understanding basic Kubernetes architecture`_
+What is Kubernetes?
+===================
+
+Kubernetes (K8s) is an open source system for automating deployment, scaling,
+and management of containerized applications. It groups containers that make
+up an application into logical units for easy management and discovery. 
+
+What is CRI+O?
+==============
+
+In addition to using Docker as a runtime, Kubernetes supports other container
+runtimes through its `Container Runtime Interface (CRI)
+<https://kubernetes.io/blog/2016/12/container-runtime-interface-cri-in-kubernetes/>`_,
+
+
+`CRI+O <https://cri-o.io/>`_  is an implementation of the Kubernetes CRI
+(Container Runtime Interface) to enable using OCI (Open Container
+Initiative) compatible runtimes. It is a lightweight alternative to using
+Docker as the runtime for kubernetes. It allows Kubernetes to use any
+OCI-compliant runtime as the container runtime for running pods. Today it
+supports runc and Kata Containers as the container runtimes but any
+OCI-conformant runtime can be plugged in principle.
 
-* `Deploying an application to your cluster`_
+
+What is Kata Containers?
+========================
+
+`Kata Containers`_ is a secure container runtime leveraging lightweight
+virtual machines that feel and perform like containers, but provide stronger
```

Comment:
Kata Containers is an alternative OCI compatible runtime that secures container workloads in a lightweight virtual machine.  It provides stronger workloads isolation using hardware virtualization technology as a second layer of defense for untrusted workloads or multi-tenant scenarios.